subiquity/controllers/filesystem.py

```diff
@@ -65,72 +120,67 @@ def __init__(self, app):
+        if probe.state == ProbeState.FAILED:
+            if not probe.restricted:
+                self._start_probe(restricted=True)
             else:
-                fname = 'probe-data.json'
-            with open(os.path.join(self.app.block_log_dir, fname), 'w') as fp:
-                json.dump(storage, fp, indent=4)
-            self.model.load_probe_data(storage)
+                if self.showing:
+                    self.start_ui()
+            return
+        if probe.restricted:
+            fname = 'probe-data-restricted.json'
+        else:
+            fname = 'probe-data.json'
+        with open(os.path.join(self.app.block_log_dir, fname), 'w') as fp:
+            json.dump(probe.result, fp, indent=4)
+        try:
+            self.model.load_probe_data(probe.result)
         except Exception:
             block_discover_log.exception(
-                "probing failed restricted=%s", restricted)
-            if not restricted:
-                block_discover_log.info("reprobing for blockdev only")
-                # Should make a crash file for apport, arrange for it to be
-                # copied onto the installed system and tell user all this
-                # happened!
-                self._reprobe()
+                "load_probe_data failed restricted=%s", probe.restricted)
+            # Should make a crash report here!
+            if not probe.restricted:
+                self._start_probe(restricted=True)
             else:
-                self._probe_state = ProbeState.FAILED
+                # OK, this is a hack
+                self._cur_probe.state = ProbeState.FAILED
                 if self.showing:
                     self.start_ui()
         else:
-            self._probe_state = ProbeState.DONE
             # Should do something here if probing found no devices.
             if self.showing:
                 self.start_ui()
 
-    def _check_probe_timeout(self):
-        log.debug("_check_probe_timeout")
-        if self._probe_state == ProbeState.PROBING:
-            log.info(
-                "unrestricted probing timed out, reprobing for blockdev only")
-            self._reprobe()
-
-    def _reprobe(self):
-        self._probe_state = ProbeState.REPROBING
-        self.run_in_bg(
-            lambda: self._bg_probe({"blockdev"}),
-            lambda fut: self._probed(fut, True),
-            )
-
     def start_ui(self):
-        if self._probe_state in [ProbeState.PROBING,
-                                 ProbeState.REPROBING]:
+        if self._cur_probe.state == ProbeState.PROBING:
             self.ui.set_body(SlowProbing(self))
-        elif self._probe_state == ProbeState.FAILED:
+        elif self._cur_probe.state == ProbeState.FAILED:
             self.ui.set_body(ProbingFailed(self))
         else:
+            self.ui_shown = True
+            # Should display a message if self._cur_probe.restricted,
+            # i.e. full device probing failed.
```

Comment:
As above... Do we have a policy for how to deal with this? Comments in general and what constitutes useful comments; when they have a chance of being addressed assuming they're potential TODOs?